test/integration/quota/quota_test.go

```diff
@@ -477,56 +473,65 @@ func TestQuotaLimitService(t *testing.T) {
-			svc:         newService("np-svc", v1.ServiceTypeNodePort, true),
-			success:     true,
-		},
-		{
-			description: "first LB type service that allocates node port should be created successfully",
-			svc:         newService("lb-svc-withnp1", v1.ServiceTypeLoadBalancer, true),
-			success:     true,
-		},
-		{
-			description: "second LB type service that allocates node port creation should fail as node port quota is exceeded",
-			svc:         newService("lb-svc-withnp2", v1.ServiceTypeLoadBalancer, true),
-			success:     false,
-		},
-		{
-			description: "first LB type service that doesn't allocates node port should be created successfully",
-			svc:         newService("lb-svc-wonp1", v1.ServiceTypeLoadBalancer, false),
-			success:     true,
-		},
-		{
-			description: "second LB type service that doesn't allocates node port creation should fail as loadbalancer quota is exceeded",
-			svc:         newService("lb-svc-wonp2", v1.ServiceTypeLoadBalancer, false),
-			success:     false,
-		},
-		{
-			description: "forth service creation should be successful",
-			svc:         newService("clusterip-svc1", v1.ServiceTypeClusterIP, false),
-			success:     true,
-		},
-		{
-			description: "fifth service creation should fail as service quota is exceeded",
-			svc:         newService("clusterip-svc2", v1.ServiceTypeClusterIP, false),
-			success:     false,
-		},
+	// Creating the first node port service should succeed
+	nodePortService := newService("np-svc", v1.ServiceTypeNodePort, true)
+	_, err = clientset.CoreV1().Services(ns.Name).Create(context.TODO(), nodePortService, metav1.CreateOptions{})
+	if err != nil {
+		t.Errorf("creating first node port Service should not have returned error: %v", err)
 	}
 
-	for _, test := range tests {
-		t.Log(test.description)
-		_, err := clientset.CoreV1().Services(ns.Name).Create(context.TODO(), test.svc, metav1.CreateOptions{})
-		if (err == nil) != test.success {
-			if err != nil {
-				t.Fatalf("Error creating test service: %v, svc: %+v", err, test.svc)
-			} else {
-				t.Fatalf("Expect service creation to fail, but service %s is created", test.svc.Name)
-			}
-		}
+	// Creating the first loadbalancer service should succeed
+	lbServiceWithNodePort1 := newService("lb-svc-withnp1", v1.ServiceTypeLoadBalancer, true)
+	_, err = clientset.CoreV1().Services(ns.Name).Create(context.TODO(), lbServiceWithNodePort1, metav1.CreateOptions{})
+	if err != nil {
+		t.Errorf("creating first loadbalancer Service should not have returned error: %v", err)
+	}
+
+	// add a delay for resource quota changes to propagate
+	time.Sleep(1 * time.Second)
+
+	// Creating another loadbalancer Service using node ports should fail because node prot quota is exceeded
+	lbServiceWithNodePort2 := newService("lb-svc-withnp2", v1.ServiceTypeLoadBalancer, true)
+	_, err = clientset.CoreV1().Services(ns.Name).Create(context.TODO(), lbServiceWithNodePort2, metav1.CreateOptions{})
```

Comment:
A delay may (will) eventually be to small and a larger delay will add unnecessary time to the test. Is it possible to do some retries?
```go
  for i := 0; i < 10; i++ {
     // add a delay for resource quota changes to propagate
     time.Sleep(1 * time.Second)
     _, err = clientset.CoreV1().Services(ns.Name).Create(context.TODO(), lbServiceWithNodePort2, metav1.CreateOptions{})
     if err == nil {
      break
     }
  }
```

Reply:
Or the delay after the first try if the normal case it that it works without a delay

Reply:
Oops, sorry, I realize that what you want to test is that the create shall succeed _after_ the constraint has been propagated. And you can't know that it has. This is a complicated test case.